File: Using Registration Model/Case 1/NotificationHubsSample.WP/NotificationService.cs
using System;
using System.Threading.Tasks;
using Microsoft.WindowsAzure.Messaging;
using Windows.Networking.PushNotifications;
using Windows.UI.Notifications;
using Windows.UI.Popups;

namespace NotificationHubsSample
{
    public class NotificationService
    {
        /// <summary>
        /// The create or update notifications async.
        /// </summary>
        public static async void CreateOrUpdateNotificationsAsync()
        {
            try
            {
                // get the channel for the app
                var pushNotificationChannel = await PushNotificationChannelManager.CreatePushNotificationChannelForApplicationAsync();
                pushNotificationChannel.PushNotificationReceived += PushNotificationChannelPushNotificationReceived;

                var hub = new NotificationHub(Constants.HubName, Constants.ConnectionString);
                
                // register the channel in NH
                
                var result = await hub.RegisterNativeAsync(pushNotificationChannel.Uri, SettingsHelper.Tags);
                if (result.RegistrationId != null)
                {
                    await ShowRegistrationIdAsync(result);
                }
            }
            catch (Exception ex)
            {
                //todo handle the exception
                // _logManager.Log(ex);
            }
        }

        /// <summary>
        /// Shows the registration identifier.
        /// </summary>
        /// <param name="result">The result.</param>
        /// <returns>Task.</returns>
        private static async Task ShowRegistrationIdAsync(Registration result)
        {
            var dialog = new MessageDialog("Registration successful: " + result.RegistrationId);
            dialog.Commands.Add(new UICommand("OK"));
            await dialog.ShowAsync();
        }

        /// <summary>
        /// Pushes the notification channel push notification received.
        /// </summary>
        /// <param name="sender">The sender.</param>
        /// <param name="args">The <see cref="PushNotificationReceivedEventArgs"/> instance containing the event data.</param>
        private static void PushNotificationChannelPushNotificationReceived(PushNotificationChannel sender, PushNotificationReceivedEventArgs args)
        {
            string result = args.NotificationType.ToString();
           
            switch (args.NotificationType)
            {
                case PushNotificationType.Badge:
                    if (args.BadgeNotification != null)
                    {
                        result += ": " + args.BadgeNotification.Content.GetXml();
                        var updater = BadgeUpdateManager.CreateBadgeUpdaterForApplication();
                        updater.Update(new BadgeNotification(args.BadgeNotification.Content));
                    }
                    break;
                case PushNotificationType.Raw:
                    if (args.RawNotification != null)
                    {
                        result += ": " + args.RawNotification.Content;
                        // todo something here...
                    }
                    break;
                case PushNotificationType.Tile:
                    if (args.TileNotification != null)
                    {
                        result += ": " + args.TileNotification.Content.GetXml();
                        var updater = TileUpdateManager.CreateTileUpdaterForApplication();
                        updater.Update(new TileNotification(args.TileNotification.Content));
                    }
                    break;
                case PushNotificationType.Toast:
                    if (args.ToastNotification != null)
                    {
                        result += ": " + args.ToastNotification.Content.GetXml();
                        // todo something here...
                    }
                    break;
                default:
                    break;
            }
            
        }
    }
}

File: Using Registration Model/Case 1/NotificationHubsSample.Win81-VB/NotificationService.vb
Imports Windows.Networking.PushNotifications
Imports Windows.UI.Notifications
Imports Microsoft.WindowsAzure.Messaging
Imports Windows.UI.Popups

Public Class NotificationService
    ''' <summary>
    ''' The create or update notifications async.
    ''' </summary>
    Public Shared Async Sub CreateOrUpdateNotificationsAsync()
        Try
            ' get the channel for the app
            Dim pushNotificationChannel = Await PushNotificationChannelManager.CreatePushNotificationChannelForApplicationAsync()
            AddHandler pushNotificationChannel.PushNotificationReceived, AddressOf PushNotificationChannelPushNotificationReceived

            Dim hub = New NotificationHub(Constants.HubName, Constants.ConnectionString)
            
            ' register the channel in NH without tags
            'Dim result = Await hub.RegisterNativeAsync(pushNotificationChannel.Uri)

            Dim result = Await hub.RegisterNativeAsync(pushNotificationChannel.Uri, SettingsHelper.Tags)

            If result.RegistrationId IsNot Nothing Then
                Await ShowRegistrationIdAsync(result)
            End If
        Catch ex As Exception
        End Try
    End Sub

    ''' <summary>
    ''' Shows the registration identifier.
    ''' </summary>
    ''' <param name="result">The result.</param>
    ''' <returns>Task.</returns>
    Private Shared Async Function ShowRegistrationIdAsync(result As Registration) As Task
        Dim dialog = New MessageDialog("Registration successful: " + result.RegistrationId)
        dialog.Commands.Add(New UICommand("OK"))
        Await dialog.ShowAsync()
    End Function

    ''' <summary>
    ''' Pushes the notification channel push notification received.
    ''' </summary>
    ''' <param name="sender">The sender.</param>
    ''' <param name="args">The <see cref="PushNotificationReceivedEventArgs"/> instance containing the event data.</param>
    Private Shared Sub PushNotificationChannelPushNotificationReceived(sender As PushNotificationChannel, args As PushNotificationReceivedEventArgs)
        Dim result As String = args.NotificationType.ToString()

        Select Case args.NotificationType
            Case PushNotificationType.Badge
                If args.BadgeNotification IsNot Nothing Then
                    result += ": " + args.BadgeNotification.Content.GetXml()
                    Dim updater = BadgeUpdateManager.CreateBadgeUpdaterForApplication()
                    updater.Update(New BadgeNotification(args.BadgeNotification.Content))
                End If
                Exit Select
            Case PushNotificationType.Raw
                If args.RawNotification IsNot Nothing Then
                    ' todo something here...
                    result += ": " + args.RawNotification.Content
                End If
                Exit Select
            Case PushNotificationType.Tile
                If args.TileNotification IsNot Nothing Then
                    result += ": " + args.TileNotification.Content.GetXml()
                    Dim updater = TileUpdateManager.CreateTileUpdaterForApplication()
                    updater.Update(New TileNotification(args.TileNotification.Content))
                End If
                Exit Select
            Case PushNotificationType.Toast
                If args.ToastNotification IsNot Nothing Then
                    ' todo something here...
                    result += ": " + args.ToastNotification.Content.GetXml()
                End If
                Exit Select
            Case Else
                Exit Select
        End Select

    End Sub
End Class
